feedback-module/src/components/common/Module.js

```diff
@@ -1,37 +1,89 @@
 import React, { useEffect, useState } from "react";
-import { GridContainer } from "@trussworks/react-uswds";
+import { GridContainer, Grid, Form } from "@trussworks/react-uswds";
 
-import { MODULE_CONTAINER_STYLE } from "../../assets/styling_classnames";
+import {
+  MODULE_CONTAINER_STYLE,
+  SCREEN_CONTAINER_STYLE,
+  H1_STYLE,
+  H1_WHITE_STYLE,
+  FORM_STYLE,
+} from "../../assets/styling_classnames";
+import { SCREENS } from "../../assets/constants";
 import Header from "./Header";
-import Screen1 from "../pages/Screen1";
-import Screen2 from "../pages/Screen2";
-import Screen3 from "../pages/Screen3";
-import Screen4 from "../pages/Screen4";
+import ModuleButton from "./Button";
+import CheckboxList from "../CheckboxList";
+import TextboxList from "../TextboxList";
 
 function Module() {
   const [feedback, setFeedback] = useState({});
+  const [userInfo, setUserInfo] = useState({});
+  const [screen, setScreen] = useState(SCREENS.feedback_type);
+  const [checkedFields, setCheckedFields] = useState(null);
+  const [otherField, setOtherField] = useState("");
+  const [inputQuestions, setInputQuestions] = useState();
 
   useEffect(() => {
+    if (screen.checkboxes) {
+      setCheckedFields(
+        screen.checkboxes.map((checkboxLabel) => {
+          return { label: checkboxLabel, checked: false };
+        })
+      );
+    }
+    if (screen.textInputs) {
+      setInputQuestions(
+        screen.textInputs.map((question) => {
+          return { question: question.text, answer: "" };
+        })
+      );
+    }
     console.log(feedback);
-  }, [feedback]);
-
-  const [screen, setScreen] = useState(
-    <Screen1
-      setFeedback={setFeedback}
-      changePage={(data) =>
-        setScreen(
-          <Screen2
-            feedbackType={data}
-            setFeedback={setFeedback}
-            changePage={() =>
-              setScreen(<Screen3 changePage={() => setScreen(<Screen4 />)} />)
-            }
-          />
-        )
-      }
-      page="[this page]"
-    />
-  );
+    console.log(userInfo);
+  }, [screen]);
+
+  const handleSubmit = (formID) => {
+    const checkedOptions =
+      checkedFields &&
+      checkedFields.map(({ label, checked }) => {
+        return {
+          checked: checked,
+          label: label === "Other" && checked ? `Other: ${otherField}` : label,
+        };
+      });
+
+    setCheckedFields(checkedOptions);
+    if (formID === "feedback") {
+      setFeedback((feedback) => {
+        feedback.checkedOptions = checkedOptions
+          .filter(({ checked }) => checked)
+          .map(({ label }) => label);
+        feedback.inputResponses = inputQuestions;
+        return feedback;
+      });
+    } else if (formID === "research") {
+      setUserInfo(inputQuestions);
+    }
+  };
+
+  const changeScreen = (text, nextScreen, feedbackID) => {
+    if (feedbackID) {
```

Comment:
Do we also want to check here if the formID is feedback because we don't want to reset all this info again in the addition pages after the feedback form. It will just unnecessarily slow the app down

Reply:
The purpose of this condition was moreso to check whether or not the button is setting the feedbackType (i.e. the buttons on the first screen). Since only the buttons on this screen have a feedbackID, this info only gets set on the first page. There's also no formID on the first page since it's just a series of buttons and not a form.